models/image.py

```diff
@@ -1120,22 +1124,18 @@ def get_upstream_provenances(self):
         This happens when the objects are used to produce, e.g., a coadd or
         a subtraction image, but they would not necessarily be loaded automatically from the DB.
         To load those products (assuming all were previously committed with their own provenances)
-        use the load_upstream_products() method.
-
-        IMPORTANT RESTRICTION:
-        When putting images in the upstream of a combined image (coadded or subtracted),
-        if there are multiple images with the same provenance, they must also have
-        loaded downstream products (e.g., SourceList) that have the same provenance.
-        This is used to maintain the ability of a downstream to recover its upstreams
-        using the provenance (which is the definition of why we need a provenance).
-        The images could still be associated with multiple different products with
-        different provenances, but not have them loaded into the relevant in-memory
-        attributes of the Image objects when creating the coadd.
-        Images from different instruments, or a coadded reference vs. a new image,
-        would naturally have different provenances, so their products could (and indeed must)
-        have different provenances. But images from the same instrument with the same provenance
-        should all be produced using the same code and parameters, otherwise it will be impossible
-        to know which product was processed in which way.
+        use the load_upstream_products() method on each of the upstream images.
+
+        IMPORTANT RESTRICTION: to maintain the ability of a downstream to recover its upstreams
+        using the provenance (which is the definition of why we need a provenance) it is not
+        allowed for different images with the same provenance to have related products
+        (e.g., a SourceList) that have different provenances.  This is because the downstream
+        would not know which SourceList to use.  Images from different instruments, or a
+        coadded reference vs. a new image, would have different provenances,
+        so their products could (and indeed must) have different provenances.
+        But images from the same instrument with the same provenance  should all
+        be produced using the same code and parameters, otherwise it will be
+        impossible to know which product was processed in which way.
```

Comment:
Ak!  Looks like this PR reverts the comment fix you put in.  Might be worth making sure that any other changes you made in review of the last PR didn't get reverted as well.

Reply:
This was the only change, and indeed I had some trouble with merging. I will bring back the text from the previous PR.